spot/plugins/SkyCam.py

```diff
@@ -36,49 +103,112 @@ def __init__(self, fv, fitsimage):
+        self.cur_data = None
+        self.img_src_name = 'Subaru Telescope (Visible)'
+
+        loaddir = ginga_home
+        path = os.path.join(loaddir, "cameras.yml")
+        if os.path.exists(path):
+            with open(path, 'r') as cam_f:
+                self.configs = yaml.safe_load(cam_f)
+        else:
+            # Sets Subaru Telescope (Visible) as default
+            # if cameras.yml does not load
+            self.config = {
+                'url': "https://allsky.subaru.nao.ac.jp/allsky/api/v0/images/download_most_recent.png",
+                'file': "allsky.png",
+                'lfile': "allsky-%ld.png",
+                'title': "Subaru Telescope (Visible)",
+                'ctr_x': 2660,
+                'ctr_y': 1850,
+                'radius': 1850,
+                'rot_deg': 9.5,
+                'flip_y': False,
+                'update_interval': 120.0}
+            self.configs = {self.img_src_name: self.config}
+        # Sets config dictionary equal to Subaru Telescope dictionary from
+        # within configs nested dictionary to start program as default
+        self.config = self.configs[self.img_src_name]
+
         # get SkyCam preferences
         prefs = self.fv.get_preferences()
         self.settings = prefs.create_category('plugin_SkyCam')
-        self.settings.add_defaults(image_center=(2660, 1850),
-                                   image_radius=1850,
-                                   image_rotation=-158.0,
-                                   #image_rotation=-20.0,
-                                   image_transform=(False, False, False),
-                                   image_update_interval=60.0)
+        self.settings.add_defaults(download_folder=tempfile.gettempdir())
+        self.update_settings()
+
         self.settings.load(onError='silent')
 
         self.ev_quit = threading.Event()
         self.sky_image_path = None
         self.flag_use_sky_image = False
+        self.flag_use_diff_image = False
 
-        self.viewer = self.fitsimage
-        self.dc = fv.get_draw_classes()
         canvas = self.dc.DrawingCanvas()
         canvas.set_surface(self.fitsimage)
         self.canvas = canvas
 
+        self.gui_up = False
+
+    # Updates settings to current image source
+    def update_settings(self):
+        self.settings.set(image_url=self.config.get('url'),
+                          image_center=(self.config.get('ctr_x'),
+                                        self.config.get('ctr_y')),
+                          image_radius=self.config.get('radius'),
+                          image_rotation=self.config.get('rot_deg'),
+                          image_transform=(self.config.get('flip_x'),
+                                           self.config.get('flip_y'),
+                                           False),
+                          image_update_interval=self.config.get(
+                                                'update_interval'))
+
```

Comment:
Suggest adding defaults for ``rot_deg`` (0.0), ``flip_x`` (False), ``flip_y`` (False) and ``update_interval`` (60.0).
This can be done like:
```python
image_rotation=self.config.get('rot_deg', 0.0)
```